packages/flutter/lib/src/material/segmented_button.dart

```diff
@@ -417,6 +549,64 @@ class SegmentedButtonState<T> extends State<SegmentedButton<T>> {
     super.dispose();
   }
 }
+
+@immutable
+class _SegmentedButtonDefaultOverlay extends MaterialStateProperty<Color?> with Diagnosticable {
```

Comment:
I don't think it's a good idea to duplicate aspects of _M3Defaults here; we want this stuff to change automatically when we regenerate _M3Defaults. Adding methods to the _M3Defaults template is OK. When you do be sure to explain the use cases in comments.